test-bridge/src/main/scala/org/scalajs/testing/bridge/JSRPC.scala

```diff
@@ -15,8 +15,15 @@ package org.scalajs.testing.bridge
 import scala.scalajs.js
 import scala.scalajs.js.annotation._
 
+/* Use the queue execution context (based on JS promises) explicitly:
+ * We do not have anything better at our disposal and it is accceptable in
+ * terms of fariness: JSRPC only handles in-between test communcation, so any
+ * future chain will "yield" to I/O (waiting for a message) or an RPC handler in
+ * a finite number of steps.
+ */
```

Comment:
This is the only place in user-affecting code where we actually use an unfair execution context. @djspiewak PTAL.

Reply:
Fascinating!